local-gpio.h

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023, Linaro Ltd.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef _LOCAL_GPIO_H_
+#define _LOCAL_GPIO_H_
+
+enum {
+	GPIO_POWER = 0,			// Power input enable
+	GPIO_FASTBOOT_KEY,		// Usually volume key
+	GPIO_POWER_KEY,			// Key to power the device
+	GPIO_USB_DISCONNECT,		// Simulate main USB connection
+	GPIO_COUNT
+};
+
+enum {
+	GPIO_ACTIVE_HIGH = 0,
+	GPIO_ACTIVE_LOW,
+};
+
+struct local_gpio_options {
+	char *gpiochip[GPIO_COUNT];
+	bool present[GPIO_COUNT];
+	unsigned int offset[GPIO_COUNT];
+	unsigned int polarity[GPIO_COUNT];
+};
+
+struct local_gpio {
+	struct local_gpio_options *options;
+	void *gpio_chip[GPIO_COUNT];
+	void *gpio_line[GPIO_COUNT];
```

Comment:
Can we please change that to have a per-GPIO struct rather than per-feature arrays?

Reply:
Yep, I had a doubt when implementing